File: CheckoutExampleObjC/CheckoutExampleObjC/ViewController.m
#import "ViewController.h"
#import "KDataCollector.h"

@implementation ViewController

- (void)viewDidLoad 
{
    [super viewDidLoad];

    switch ([[KDataCollector sharedCollector] environment]) {
        case KEnvironmentProduction:
            self.environment.text = @"Production";
            break;
        case KEnvironmentTest:
            self.environment.text = @"Test";
            break;
        default:
            self.environment.text = @"Unknown";
    }
    
    self.merchant.text = [NSString stringWithFormat:@"%06ld", [[KDataCollector sharedCollector] merchantID]];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:@"Checkout"]) {
        UINavigationController *navigationController = segue.destinationViewController;
        //To capture Analytics data starting from iOS version 13.0 or later, the modal presentation style should be fullscreen.
        navigationController.modalPresentationStyle = UIModalPresentationFullScreen;
        CheckoutViewController *controller = [navigationController viewControllers][0];
        controller.delegate = self;
    }
}

- (void)checkoutViewControllerDidFinish:(CheckoutViewController *)controller 
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: CheckoutExampleSwift/CheckoutExampleSwift/ViewController.swift
import UIKit
import KountSDK

class ViewController: KountAnalyticsViewController, CheckoutViewControllerDelegate  {

    @IBOutlet weak var merchant: UILabel?
    @IBOutlet weak var environment: UILabel?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        merchant!.text = KDataCollector.shared.merchantID
        
        switch KDataCollector.shared.environment {
        case KountEnvironment.test:
            environment!.text = "Test"
        case KountEnvironment.production:
            environment!.text = "Production"
        default:
            environment!.text = "Unknown"
        }
    }

    func didFinish(_ sender: CheckoutViewController) {
        self.dismiss(animated: true, completion: nil)
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "Checkout" {
            let navigationController = segue.destination
            //To capture Analytics data starting from iOS version 13.0 or later, the modal presentation style should be fullscreen.
            navigationController.modalPresentationStyle = .fullScreen
            let controller :CheckoutViewController = navigationController.childViewControllers[0] as! CheckoutViewController
            controller.checkoutDelegate = self
        }
        
    }
}

